cpp/src/cylon/ops/kernels/set_kernel.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cylon/ops/kernels/prepare_array.hpp>
+#include <cylon/ctx/arrow_memory_pool_utils.hpp>
+#include <cylon/ops/kernels/set_kernel.hpp>
+
+namespace cylon {
+namespace kernel {
+
+template<Status op(const std::shared_ptr<Table> &, const std::shared_ptr<Table> &, std::shared_ptr<Table> &)>
```

Comment:
this can simply be, 
```suggestion
template<typename op>
```

Reply:
I feel thhe above code is more readable as someone looking at the code knows what he is getting